### PR TITLE
[SERVICE-572] Notifications Offline-mode

### PR DESCRIPTION
**Pull request by ngunday**
Implements offline-mode for notifications service. Caches the app via a service worker when the app runs on online mode and serves the cached files when system is offline.

### CODE REVIEW

File: src/provider/ServiceWorker.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    './ui/css/AkkuratPro/AkkBd_Pro_1.otf',
+    './ui/css/AkkuratPro/AkkIt_Pro_1.otf',
+    './ui/css/AkkuratPro/AkkLgIt_Pro_1.otf',
+    './ui/css/AkkuratPro/AkkLg_Pro_1.otf',
+    './ui/css/AkkuratPro/AkkRg_Pro_1.otf',
+    './ui/css/themes/dark.css',
+    './ui/css/themes/light.css',
+    './ui/image/shapes/arrow.png',
+    './ui/image/shapes/notifications-x.png',
+    './ui/image/shapes/trayIcon.png'
```
Comment (ryanpetersonOF) [inline src/provider/ServiceWorker.js]: Is there a way to cache everything under `ui/*`?

**Comment by ryanpetersonOF** [comment]
I dont know if this belongs as part of this ticket, but are there any considerations for caching application icons that are placed in within the notifications?

File: src/provider/ServiceWorker.js (hunk 1 above)
Comment (ngunday) [reply src/provider/ServiceWorker.js]: I think I have a better solution to address both your comment. SW now caches all the fetched files dynamically and serves them if they exist.

File: src/provider/ServiceWorker.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Set a name for 'he current cache
```
Comment (ryanpetersonOF) [inline src/provider/ServiceWorker.js]: Minor spelling mistake

**Pull request merged**